src/main/java/com/sucy/skill/listener/ShieldBlockListener.java

```diff
@@ -0,0 +1,43 @@
+package com.sucy.skill.listener;
+
+import com.sucy.skill.api.event.PlayerBlockDamageEvent;
+import org.bukkit.Bukkit;
+import org.bukkit.Statistic;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.player.PlayerStatisticIncrementEvent;
+
+import java.util.HashMap;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Listener to throw custom {@link PlayerBlockDamageEvent}
+ */
+public class ShieldBlockListener extends SkillAPIListener {
+
+    private final HashMap<Player, EntityDamageByEntityEvent> volatileMap = new HashMap<>();
+
+    /**
+     * Remember EntityDamageByEntityEvent in a moment
+     */
+    @EventHandler
+    public void onDamage(EntityDamageByEntityEvent event){
+        if (!(event.getEntity() instanceof Player)) return;
+        Player player = (Player) event.getEntity();
+        volatileMap.put(player, event);
+        new Timer().schedule(new TimerTask() {@Override public void run() {volatileMap.remove(player);}},100);
```

Comment:
I assume this timer is needed because it could be possible that the stat event is never called, correct?